packages/gatsby-plugin-functions/src/gatsby-node.ts

```diff
@@ -0,0 +1,88 @@
+import fs from "fs-extra"
+import glob from "globby"
+import path from "path"
+import webpack from "webpack"
+
+import { ParentSpanPluginArgs, PluginOptions } from "gatsby"
+
+export async function onPreInit(
+  { reporter }: ParentSpanPluginArgs,
+  { path: functionsDirectoryPath }: PluginOptions
+): Promise<void> {
+  const activity = reporter.activityTimer(`building functions`)
+  activity.start()
+
+  const functionsDirectory = path.resolve(
+    process.cwd(),
+    functionsDirectoryPath as string
+  )
+
+  const functionsGlob = `**/*.{js,ts}`
+
+  // Get initial list of files
+  const files = await glob(functionsGlob, { cwd: functionsDirectory })
+
+  // console.log({ files })
+
+  const knownFunctions = new Set(files)
+
+  // console.log({ knownFunctions })
+
+  // if (![`develop`, `build-javascript`].includes(stage)) {
+  //   return Promise.resolve()
+  // }
+
+  await fs.ensureDir(path.join(process.cwd(), `.cache`, `functions`))
+
+  await fs.emptyDir(path.join(process.cwd(), `.cache`, `functions`))
+
+  // console.log({
+  //   path: path.join(functionsDirectory, `month.js`),
+  // })
+
+  await Promise.all(
+    files.map(file => {
+      const config = {
+        entry: path.join(functionsDirectory, file),
+        output: {
+          path: path.join(process.cwd(), `.cache`, `functions`),
```

Comment:
Rather than resolving this directory name every time, why not do it once up the top and assign it to a variable

Reply:
Yup, this was a quick hacky first pass. I’ll definitely be cleaning all of this sort of stuff up.